File: src/Java/InfoVis/PassThrough.java
import vtk.vtkNativeLibrary;
import vtk.vtkPassThrough;
import vtk.vtkSphereSource;

public class PassThrough 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
  
    // Create a sphere
    vtkSphereSource sphereSource = new vtkSphereSource();
    sphereSource.Update();
    
    System.out.println("Points before:" + " " + sphereSource.GetOutput().GetNumberOfPoints());
    
    vtkPassThrough passThrough = new vtkPassThrough();
    passThrough.SetInputConnection(sphereSource.GetOutputPort());
    passThrough.Update();
    passThrough.GetOutput();

    System.out.println("Points after:" + " " + sphereSource.GetOutput().GetNumberOfPoints());
   
  }
}

File: src/Cxx/InfoVis/PassThrough.cxx
#include <vtkSmartPointer.h>
#include <vtkSphereSource.h>
#include <vtkPolyData.h>
#include <vtkPassThrough.h>

int main(int, char *[])
{
  // Create a sphere
  vtkSmartPointer<vtkSphereSource> sphereSource = 
    vtkSmartPointer<vtkSphereSource>::New();
  sphereSource->Update();
  
  std::cout << "Points before: " << sphereSource->GetOutput()->GetNumberOfPoints() << std::endl;
  
  vtkSmartPointer<vtkPassThrough> passThrough = 
    vtkSmartPointer<vtkPassThrough>::New();
  passThrough->SetInputConnection(sphereSource->GetOutputPort());
  passThrough->Update();
  
  vtkSmartPointer<vtkPolyData> output = dynamic_cast<vtkPolyData*>(passThrough->GetOutput());
  
  std::cout << "Points after: " << output->GetNumberOfPoints() << std::endl;
  
  return EXIT_SUCCESS;
}
